HexView.Wpf/ExtensionMethods.cs

```diff
@@ -0,0 +1,129 @@
+        /// </summary>
+        /// <param name="reader">in-use BinaryReader.</param>
+        /// <returns>Int16 (signed short).</returns>
+        public static short ReadInt16BE(this BinaryReader reader)
+        {
+            var data = reader.ReadBytes(2);
+            Array.Reverse(data);
+            return BitConverter.ToInt16(data, 0);
+        }
+
+        /// <summary>
+        /// Reads a 2-byte unsigned integer in big-endian order and advances the current position of the stream by 2 bytes.
+        /// </summary>
+        /// <param name="reader">in-use BinaryReader.</param>
+        /// <returns>UInt16 (unsigned short).</returns>
+        public static ushort ReadUInt16BE(this BinaryReader reader)
+        {
+            var data = reader.ReadBytes(2);
+            Array.Reverse(data);
+            return BitConverter.ToUInt16(data, 0);
+        }
+
+        /// <summary>
+        /// Reads a 4-byte signed integer in big-endian order and advances the current position of the stream by 4 bytes.
+        /// </summary>
+        /// <param name="reader">in-use BinaryReader.</param>
+        /// <returns>Int32 (signed int).</returns>
+        public static int ReadInt32BE(this BinaryReader reader)
+        {
+            var data = reader.ReadBytes(4);
+            Array.Reverse(data);
+            return BitConverter.ToInt32(data, 0);
+        }
+
+        /// <summary>
+        /// Reads a 4-byte unsigned integer in big-endian order and advances the current position of the stream by 4 bytes.
+        /// </summary>
+        /// <param name="reader">in-use BinaryReader.</param>
+        /// <returns>UInt32 (unsigned int).</returns>
+        public static uint ReadUInt32BE(this BinaryReader reader)
+        {
+            var data = reader.ReadBytes(4);
+            Array.Reverse(data);
+            return BitConverter.ToUInt32(data, 0);
+        }
+
+        /// <summary>
+        /// Reads a 8-byte signed integer in big-endian order and advances the current position of the stream by 8 bytes.
+        /// </summary>
+        /// <param name="reader">in-use BinaryReader.</param>
+        /// <returns>Int64 (signed long).</returns>
+        public static long ReadInt64BE(this BinaryReader reader)
+        {
+            var data = reader.ReadBytes(8);
+            Array.Reverse(data);
+            return BitConverter.ToInt64(data, 0);
+        }
+
+        /// <summary>
+        /// Reads a 8-byte unsigned integer in big-endian order and advances the current position of the stream by 8 bytes.
+        /// </summary>
+        /// <param name="reader">in-use BinaryReader.</param>
+        /// <returns>UInt64 (unsigned long).</returns>
+        public static ulong ReadUInt64BE(this BinaryReader reader)
+        {
+            var data = reader.ReadBytes(8);
+            Array.Reverse(data);
+            return BitConverter.ToUInt64(data, 0);
+        }
```

Comment:
This implementation is going to be quite slow with the use of arrays. When rendering large amounts of data this will have noticeable effects. In one of my other projects I actually have an `EndianBinaryReader` already implemented. You can find it here:

https://github.com/fjeremic/BinaryTools.Elf/blob/master/BinaryTools.Elf/Io/EndianBinaryReader.cs

Actually in that directory you will find much of the same APIs as in your PR:
https://github.com/fjeremic/BinaryTools.Elf/tree/master/BinaryTools.Elf/Io

I'm wondering if we could just pull from there. For example your `ReadUInt64BE` is implemented here:
https://github.com/fjeremic/BinaryTools.Elf/blob/master/BinaryTools.Elf/Io/EndianBinaryReader.cs#L153

Which delegates to this call:
https://github.com/fjeremic/BinaryTools.Elf/blob/master/BinaryTools.Elf/Io/EndianBitConverter.cs#L94

The reason this is much faster is because the JIT compiler will recognize this pattern and turn it into `bswap` x86 instruction for example which is as fast as it gets.

Do you think it is feasible to borrow from the aforementioned project?

Reply:
Filip, please use your faster code for this.  See additional comment below. (I didn't see spot for commenting here...)